File: src/main/java/ca/joyfactory/coims/domain/Role.java
package ca.joyfactory.coims.domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by Joinsu on 2017-10-16.
 */
@Entity
public class Role {

    @Id
    @Column( name = "ROLE_ID")
    @GeneratedValue( strategy = GenerationType.AUTO)
    private long id;

    @Column( length = 20, nullable = false)
    @Enumerated( EnumType.STRING)
    private RoleType role;

    @ManyToMany( mappedBy = "roles")
    @JsonIgnoreProperties( "roles")
    private List<User> users = new ArrayList<User>();

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public RoleType getRole() {
        return role;
    }

    public void setRole(RoleType role) {
        this.role = role;
    }

    public List<User> getUsers() {
        return users;
    }

    public void setUsers(List<User> users) {
        this.users = users;
    }
}

File: frontend/src/app/_model/role.ts
/**
 * Created by Joinsu on 2018-07-31.
 */
export class Role{
  private id: number;
  private role: string;

  constructor( id: number, role: string){
    this.id = id;
    this.role = role;
  }
  getId(): number {
    return this.id;
  }

  setId(value: number) {
    this.id = value;
  }

  getRole(): string {
    return this.role;
  }

  setRole(value: string) {
    this.role = value;
  }
}
